packages/plugins/src/plugins/diagnostic-fix.plugin.ts

```diff
@@ -34,13 +34,19 @@ export class DiagnosticFixPlugin extends Plugin {
         continue;
       }
 
-      const fix = codefixes.getFixForDiagnostic(diagnostic);
-      const fixedFiles: FixedFile[] = fix?.run(diagnostic, this.service) || [];
-      const fixed = fixedFiles.length > 0;
+      let fixedFiles: FixedFile[] = [];
+
+      const fixes = codefixes.getCodeFixes(diagnostic);
+
+      while (fixes.length > 0 && fixedFiles.length === 0) {
+        fixedFiles = fixes.shift()?.run(diagnostic, this.service) || [];
```

Comment:
This is considered a hack because we currently can't predict if FixTransform will be applied or not, so we have to try to apply a fix and check the result to know if it was used. This code will be cleaned out after Transforms is replaced with CodeFixActions in the next iteration, so getCodeFixes will be returned only one CodeFixAction.